library/src/main/java/com/eyeem/deviceinfo/Factory.java

```diff
@@ -233,4 +248,16 @@ private static DisplayMetrics getDisplayMetrics(Context context) {
       dm.setToDefaults();
       return dm;
    }
+
+   @RequiresPermission(Manifest.permission.USE_FINGERPRINT)
+   private static boolean isFingerPrintAvailable(Context context){
+      // Check if we're running on Android 6.0 (M) or higher
+      if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+         //Fingerprint API only available on from Android 6.0 (M)
+         FingerprintManager fingerprintManager = (FingerprintManager) context.getSystemService(Context.FINGERPRINT_SERVICE);
+         return fingerprintManager.isHardwareDetected();
```

Comment:
I'm thinking on this part here. I'm kind of not convinced about it.
I would have to test it more, but I believe it might crash if running in Marshmallow on an application that does not have fingerprint permission.
So, maybe we should be check `if( permissions granted)` as well ?